storefront/components/SidebarMenu/SidebarMenu.tsx

```diff
@@ -2,47 +2,96 @@ import React from 'react';
 import Link from 'next/link';
 import cn from 'classnames';
 
-import { Tag } from '../Tag/Tag';
-import { capitalizeString } from '../../utils/StringHelpers';
+import {
+  capitalizeString,
+  convertQueryToReadable,
+  removeStringExtension,
+  convertMenuItemToRelativePath,
+  replaceBackSlashWithForwardSlash,
+} from '../../utils/StringHelpers';
+import type {
+  PageMenuDataType,
+  PageMenuItemType,
+} from '../../utils/menus/PageMenu';
 
 import classes from './SidebarMenu.module.css';
 
 interface SidebarMenuProps {
   title: string;
-  items: [];
+  menu: PageMenuDataType;
+  activeRouterPath: string;
 }
 
-interface SidebarMenuItem {
-  active: boolean;
-  url: string;
-  status: string;
-  title: string;
-}
+const SidebarMenu = ({ title, menu, activeRouterPath }: SidebarMenuProps) => {
+  /* Convert name given from MenuTree to a readable name */
+  const getListItemName = (name: string) => {
+    return convertQueryToReadable(
+      capitalizeString(removeStringExtension(name)),
+    );
+  };
+  /* Convert path given from MenuTree to relative path with no file extension */
+  const getListItemPath = (path: string) => {
+    return convertMenuItemToRelativePath(
+      removeStringExtension(replaceBackSlashWithForwardSlash(path)),
+    );
+  };
+  /* Check if item path in menu is equal to the activePath sent in to the component */
+  const isItemActive = (path: string) => {
+    return getListItemPath(path) == activeRouterPath;
+  };
```

Comment:
Not that it has to big of an performance impact, but its recommended to have helper functions like these outside of the render function (or wrapped in a `useCallback`). Its either way good a good habit to start having :) Suggest moving these above `SidebarMenu`